includes/ui/Notice.php

```diff
@@ -0,0 +1,226 @@
+<?php
+/**
+ * Notice component for the frontend.
+ *
+ * @package wp-job-manager
+ * @since $$next-version$$
+ */
+
+namespace WP_Job_Manager\UI;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Notice component for the frontend.
+ *
+ * @since $$next-version$$
+ *
+ * @internal This API is still under development and subject to change.
+ * @access private
+ */
+class Notice {
+
+	/**
+	 * A success message.
+	 *
+	 * @param string|array $args A message, or an array of options accepted by Notice::render.
+	 */
+	public static function success( $args ) {
```

Comment:
As discussed in my own PR, using static methods everywhere is the same with using a procedural style. I think that we should use normal methods and instances if we want the benefits of OOP. 

There are also plenty of discussions on the matter online, for example [here](https://stackoverflow.com/questions/33705976/when-should-i-use-static-methods) and [here](https://stackoverflow.com/questions/47118926/should-i-avoid-using-too-many-static-methods-in-php).

Reply:
What would be the benefits of an OOP approach here? The downside is it's more boilerplate code and a harder to use API, replacing `echo Notice::success( $args )` with something like
```php
$notice = new Success_Notice( $args );
echo $notice->render();
```

Reply:
Sorry didn't clarify that but I was not suggesting that an OOP approach is better than procedural here, what I was trying to say was that if we use a class it becomes OOP otherwise we should use functions.

There are benefits of using instance methods instead of static ones though, as static methods cannot be mocked AFAIK and they are not as easily overriden. You can always do `(new Success_Notice()->render() ) if you want to call render in a single expression.